### PR TITLE
Changes to the build output directories

### PR DESCRIPTION
**Pull request by dsmolars**
* Each configuration + platform + architecture configuration now has
  it's own output directory
* Each configuration + platform + architecture configuration now has
  it's own cmake files output directory

Fixes: #369

Change-Id: Id6ee2a227c4cdfb1ede55cd3ef2a129b0ea1b0b1

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -85,12 +85,17 @@ configure_file("${CMAKE_SOURCE_DIR}/project_version.h.in" "${CMAKE_BINARY_DIR}/p
...
-include("Test")
+include("Cpp")
 include("Lint")
+include("Test")
 include("FindVulkanVersion")
 
 # Apply misc build directives to the given target
 macro(common_build_directives TARGET)
+    if(MSVC)
+        target_compile_options(${TARGET} PRIVATE $<$<CONFIG:Debug>:/MTd> $<$<CONFIG:Release>:/MT>)
```
Comment (dustin-lunarg) [inline CMakeLists.txt:96]: This appears to be generating linker warnings with Release builds and errors with Debug builds.  If you remove the target_compile_options addition, I think this should be good to merge.

**Comment by dsmolars** [comment]
I'll try to repro this, but I don't remember seeing this on the amd-builds.

File: CMakeLists.txt (hunk 1 above)
Comment (dsmolars) [reply CMakeLists.txt:96]: I'll try to repro this, but I don't remember seeing this on the amd-builds.

Comment (dsmolars) [reply CMakeLists.txt:96]: OK, found the issue.
For some reason I can only repro on my test system - I suspect it's related to what is installed on the system.
The issue is that LZ4 static lib is linked against the dynamic runtime (/MD).

Comment (dsmolars) [reply CMakeLists.txt]: Removed the /MT and /MTd options when compiling with MSVC

**Pull request merged**